src/coverlet.core/Helpers/RetryHelper.cs

```diff
@@ -54,7 +55,12 @@ public T Do<T>(
           }
           return action();
         }
-        catch (Exception ex)
+        catch (DirectoryNotFoundException)
+        {
+          // do nothing
+          return default;
```

Comment:
Sorry I don't have that much time to look into everything but in which scenario we don't want to throw the exception? E.g. if someone/something deletes the temp directory with the orginal modules before we could restore them? Is this something we can just ignore because the user will be notified in another place?

Reply:
This will suppress a retry for a not existing folder or not accessible folder. I had this situation which finally makes any retry useless. 

By the way, this code was copied a long time ago (2017) from stackoverflow. Newer comments states that this implementation is buggy and **Polly** was recommended as an alternative.